extensions/machine-learning-services/src/modelManagement/azureModelRegistryService.ts

```diff
@@ -263,16 +286,29 @@ export class AzureModelRegistryService {
 		return baseUri;
 	}
 
-	private async getClient(baseUri: string, account: azdata.Account, subscription: azureResource.AzureResourceSubscription, tenant: any): Promise<AzureMachineLearningWorkspaces> {
-		const tokens = await this._apiWrapper.getSecurityToken(account, azdata.AzureResource.ResourceManagement);
-		const token = tokens[tenant.id].token;
-		const tokenType = tokens[tenant.id].tokenType;
-		const options: AzureMachineLearningWorkspacesOptions = {
-			baseUri: baseUri
-		};
-		const client = new AzureMachineLearningWorkspaces(new TokenCredentials(token, tokenType), subscription.id, options);
-		client.apiVersion = this._config.amlApiVersion;
-		return client;
+	private getModelClient(amlClient: AzureMachineLearningWorkspaces) {
+		return this._modelClient ?? new WorkspaceModels(amlClient);
+	}
+
+	private async getAmlClient(
+		account: azdata.Account,
+		subscription: azureResource.AzureResourceSubscription,
+		tenant: any,
+		options: AzureMachineLearningWorkspacesOptions | undefined = undefined,
+		apiVersion: string | undefined = undefined): Promise<AzureMachineLearningWorkspaces> {
+		if (this._amlClient) {
+			return this._amlClient;
+		} else {
+			const tokens = await this._apiWrapper.getSecurityToken(account, azdata.AzureResource.ResourceManagement);
+			const token = tokens[tenant.id].token;
```

Comment:
Is it guaranteed that the tokens will have one with the specified tenant at this point?